File: benchmarks/python/b2_bubble_sort.py
import random
N = 100_000; random.seed(42)
arr = [random.randint(0, N) for _ in range(N)]
for i in range(N - 1):
    swapped = False
    for j in range(N - i - 1):
        if arr[j] > arr[j+1]: arr[j], arr[j+1] = arr[j+1], arr[j]; swapped = True
    if not swapped: break
print(arr[0], arr[-1])

File: benchmarks/go/b2_bubble_sort.go
package main
import ("fmt"; "math/rand")
func main() {
    N := 100000
    r := rand.New(rand.NewSource(42))
    arr := make([]int, N)
    for i := range arr { arr[i] = r.Intn(N) }
    for i := 0; i < N-1; i++ {
        swapped := false
        for j := 0; j < N-i-1; j++ {
            if arr[j] > arr[j+1] { arr[j], arr[j+1] = arr[j+1], arr[j]; swapped = true }
        }
        if !swapped { break }
    }
    fmt.Println(arr[0], arr[N-1])
}
